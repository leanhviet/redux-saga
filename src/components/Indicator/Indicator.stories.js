// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import Indicator from './index'

storiesOf('Indicator', module).add('Default', () => <Indicator />)
