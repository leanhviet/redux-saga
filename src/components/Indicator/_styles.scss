.indicator {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  z-index: 1;
  opacity: 0.5;
  background-color: $black;

  &__img {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 25px;
    height: 25px;
    transform: -50%, -50%;
    border: 5px solid $white;
    animation: spin 2s linear infinite;
    display: inline-block;

    border: {
      top: 5px solid $red;
      radius: 50%;
    }
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
