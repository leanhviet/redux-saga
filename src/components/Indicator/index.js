import React from 'react'

const Indicator = () => (
  <div className="indicator">
    <div className="indicator__img" />
  </div>
)

export default Indicator
