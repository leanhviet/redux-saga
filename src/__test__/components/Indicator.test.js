// Libs
import React from 'react'
import { shallowToJson } from 'enzyme-to-json'
import { shallow } from 'enzyme'

// Component
import Indicator from '../../components/Indicator'

describe('Indicator component', () => {
  // Snapshot
  test('should render correctly Indicator component default props', () => {
    const component = shallow(<Indicator />)

    expect(shallowToJson(component)).toMatchSnapshot()
  })
})
