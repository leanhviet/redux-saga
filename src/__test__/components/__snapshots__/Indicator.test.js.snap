// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator component should render correctly Indicator component default props 1`] = `
<div
  className="indicator"
>
  <div
    className="indicator__img"
  />
</div>
`;
